tests/unit/test_charm.py

```diff
@@ -107,3 +108,112 @@ def test_traefik_integration(harness_server_name_configured: Harness) -> None:
         "port": str(SYNAPSE_PORT),
         "strip-prefix": "true",
     }
+
+
+@pytest.mark.parametrize("harness", [0], indirect=True)
+def test_server_name_change(harness_server_name_configured: Harness) -> None:
+    """
+    arrange: start the Synapse charm, set Synapse container to be ready and set server_name.
+    act: change to a different server_name.
+    assert: Synapse charm should prevent the change with a BlockStatus.
+    """
+    harness = harness_server_name_configured
+    server_name_changed = "pebble-layer-1.synapse.com"
+    harness.update_config({"server_name": server_name_changed})
+    assert isinstance(harness.model.unit.status, ops.BlockedStatus)
+    assert "is different from the existing" in str(harness.model.unit.status)
+
+
+@pytest.mark.parametrize("harness", [0], indirect=True)
+def test_reset_instance_action(harness_server_name_configured: Harness) -> None:
+    """
+    arrange: start the Synapse charm, set Synapse container to be ready and set server_name.
+    act: run reset-instance action.
+    assert: Synapse charm should reset the instance.
+    """
+    harness = harness_server_name_configured
+    harness.set_leader(True)
+    server_name_changed = "pebble-layer-1.synapse.com"
+    harness.update_config({"server_name": server_name_changed})
+    assert isinstance(harness.model.unit.status, ops.BlockedStatus)
+    assert "is different from the existing" in str(harness.model.unit.status)
+    event = unittest.mock.Mock()
+    # Calling to test the action since is not possible calling via harness
+    harness.charm._reset_instance_action(event)  # pylint: disable=protected-access
+    assert event.set_results.call_count == 1
+    event.set_results.assert_called_with({"reset-instance": True})
+    assert isinstance(harness.model.unit.status, ops.ActiveStatus)
+
+
+@pytest.mark.parametrize("harness", [1], indirect=True)
+def test_reset_instance_action_failed(harness_server_name_configured: Harness) -> None:
+    """
+    arrange: start the Synapse charm, set Synapse container to be ready and set server_name.
+    act: change server_name and run reset-instance action.
+    assert: Synapse charm should be blocked by error on migrate_config command.
+    """
+    harness = harness_server_name_configured
+    harness.set_leader(True)
+    server_name_changed = "pebble-layer-1.synapse.com"
+    harness.update_config({"server_name": server_name_changed})
+    assert isinstance(harness.model.unit.status, ops.BlockedStatus)
+    assert "is different from the existing" in str(harness.model.unit.status)
+    event = unittest.mock.Mock()
+    # Calling to test the action since is not possible calling via harness
+    harness.charm._reset_instance_action(event)  # pylint: disable=protected-access
```

Comment:
I've seen in other charms that `# pylint: disable=protected-access` is at the top of the module as a module scoped disable, would this be helpful here?

Reply:
Nice, better than repeating, changed!